x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -354,6 +358,17 @@ public Map<String, Object> domainUsageStats() {
         }
     }
 
+    @Override
+    protected void doStart() {}
+
+    @Override
+    protected void doStop() {}
+
+    @Override
+    protected void doClose() throws IOException {
+        IOUtils.close(allConfiguredRealms.stream().filter(r -> r instanceof Releasable).map(r -> (Releasable) r).toList());
```

Comment:
Should this be?
```suggestion
        IOUtils.close(allConfiguredRealms.stream().filter(r -> r instanceof Closeable).map(r -> (Closeable) r).toList());
```

Every `Releasable` is also a `Closeable`, and `IOUtils` only needs `Closeable`.

If we only support `Releasable` then we risk someone implementing `Closeable` and being confused about why it doesn't work. So, in that case we probably should make it invalid to implement `Closeable` without implementing `Releaseable`.

Or, we can just allow `Closeable`.

Reply:
Yes, that's correct, it should close all `Closeable`s, thanks for the suggestion.